wolfcrypt/src/port/atmel/atmel.c

```diff
@@ -896,12 +909,80 @@ int atcatls_verify_signature_cb(WOLFSSL* ssl, const byte* sig, unsigned int sigS
     return ret;
 }
 
+static int atcatls_set_certificates(WOLFSSL_CTX *ctx) 
+{
+    int ret = 0;
+    ATCA_STATUS status;
+    size_t signerCertSize = 0;
+    uint8_t signerCert[signerCertSize];
```

Comment:
This array is sized as zero...  Typically this is 521 bytes for the call to `tng_atcacert_read_signer_cert`.

Reply:
ah! wrote this code a while back and moved all declarations to the top foolishly while tired 😢 

since we are hard coding the size,  do you think it is better to make it 1024 to be future proof if some info changes in the signer/device certificates in the future?

Reply:
Thanks, much better. However some compilers do not like a non-const variable as the size for an array. You'll need to put 1024 explicitly or add a macro define for it. You might also consider reducing this a bit. Apparently the 521 and 552 values are ATECC fixed...

```
#ifndef ATCATLS_MAX_CERT_SIZE
#define ATCATLS_MAX_CERT_SIZE 1024
#endif

 size_t signerCertSize = ATCATLS_MAX_CERT_SIZE;
uint8_t signerCert[ATCATLS_MAX_CERT_SIZE];

...

char devCertChain[ATCATLS_MAX_CERT_SIZE*2];
```

Reply:
ok. Should this be put in the header / start of the C file or can it be within the function block?